examples/text-generation/run_lm_eval.py

```diff
@@ -214,6 +214,32 @@ def _model_call(self, inps: torch.Tensor) -> torch.Tensor:
         logits = logits.to(torch.float32)
         return logits
 
+    def get_model_info(self) -> dict:
+        """
+        Mod from method to get Hugging Face model information for experiment reproducibility.
+        """
+
+        def get_model_num_params(model) -> int:
+            if hasattr(model, "num_parameters"):
+                return model.num_parameters()
+            if hasattr(model, "parameters"):
+                return sum(p.numel() for p in model.parameters())
+            else:
+                return -1
+
+        def get_model_dtype(model) -> str:
+            if hasattr(model, "dtype"):
+                return model.dtype
+            else:
+                return ""
```

Comment:
```suggestion
            elif hasattr(model, "parameters"):
                return next(model.parameters()).dtype
            else:
                return ""
```

what do you think to get the data type of the model's parameters? just a thought

Reply:
Good thought